### PR TITLE
Develop script `GarbledTextValidation.cs`, `MissingContentValidation.cs` and `TestPageContent.cs`

### PR DESCRIPTION
**Pull request by zimuli157**
1.
**Testcase:**
Garbled text - Develop script `GarbledTextValidation.cs`.

**Design:**

- Create a browser instance.
- Fetch all `p` tags.
- Loop through each `p` tag.
- Check if the text is garbled, if so, return the garbled text.

**Result:**
[link](https://learn.microsoft.com/en-us/python/api/azure-communication-rooms/azure.communication.rooms.roomsclient?view=azure-python)
image
image

2.

**Testcase:**
Missing content. - Develop script `MissingContentValidation.cs`.

**Design:**

- Create a browser instance.
- Fetch all `table` tags and `div` tags containing `data-heading-level`.
- Fetch all `tr` tags in the `table` tag.
- Loop through each `tr` tag and get the `div` tag closest to it.
- Check if there is an empty data cell, If there is an empty cell, output error information.

**Result:**
[link](https://learn.microsoft.com/en-us/python/api/azure-batch/azure.batch?view=azure-python)
image

@zedy-wj and @v-xuto for notification.

### CODE REVIEW

File: ContentValidation/GarbledTextValidation.cs [hunk 1]
```diff
@@ -12,63 +12,40 @@ public GarbledTextValidation(IPlaywright playwright)
...
-                }
 
+                //Check if the text is garbled, if so, return the garbled text
                 if (Regex.IsMatch(text, @":[\w]+(?:\s+[\w]+){0,2}:"))
                 {
                     res.Result = false;
-                    var errorMessage = $" \n{text} " + $"\nLink : {testLink}" + $"{specificAnchorHref}\n";
-                    errorMessages.Add(errorMessage);
+                    res.NumberOfOccurrences += 1;
+                    numbers += 1;
```
Comment (zedy-wj) [inline ContentValidation/GarbledTextValidation.cs:38]: The usage of `numbers` is the same as `res.NumberOfOccurrences`, so there is no need to define it again.

File: ContentValidation/GarbledTextValidation.cs [hunk 2]
```diff
@@ -12,63 +12,40 @@ public GarbledTextValidation(IPlaywright playwright)
...
+                    errorMessage += $"\n{numbers}.\n" + text;
                 }
             }
-            res.ErrorMsg = string.Join("",errorMessages);
+
+            res.ErrorLink = $"\nError Link: {testLink}\n";
+            res.ErrorInfo = "Error Info: The testLink have Garbled Text.\n";
```
Comment (zedy-wj) [inline ContentValidation/GarbledTextValidation.cs:44]: Update to: "Error Info: The test link has garbled text.\n "

File: ContentValidation/MissingContentValidation.cs [hunk 3]
```diff
@@ -53,15 +61,20 @@ public async Task<TResult> Validate(string testLink)
...
                 }
 
+                //Check if there is an empty data cell, if so, return the nearest link.
                 if (string.IsNullOrWhiteSpace(textContent))
                 {
                     res.Result = false;
-                    var errorMessage = $"{textContent} " + $"\nLink : {testLink}+{specificAnchorHref}";
-                    errorMessages.Add(errorMessage);
+                    res.NumberOfOccurrences += 1;
+                    numbers += 1;
```
Comment (zedy-wj) [inline ContentValidation/MissingContentValidation.cs:69]: The same as other comment.

Comment (zimuli157) [reply ContentValidation/MissingContentValidation.cs]: Updated

**Comment by v-xuto** [comment]
@zimuli157 If this PR is not needed, please close it.

**Pull request closed without merging**